plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizardModel.java

```diff
@@ -253,6 +256,14 @@ public void setProject(IProject project) {
 		firePropertyChange(PROPERTY_PROJECT, this.project, this.project = project);
 	}
 
+	public void setProjectOrDefault(IProject project) {
+		if (project != null) {
+			setProject(project);
```

Comment:
May project be validated in projects? If the projects list has no project with that name, then default project should be set. Just obtained by a request to the connection, the list may have an updated instance of the project, then it should be set instead of the original one.